ledger-tool/src/ledger_utils.rs

```diff
@@ -404,7 +404,14 @@ pub fn load_and_process_ledger(
     .map_err(LoadAndProcessLedgerError::ProcessBlockstoreFromRoot);
 
     exit.store(true, Ordering::Relaxed);
-    accounts_background_service.join().unwrap();
+    // Typically, we would want to join all threads before returning. However,
+    // AccountsBackgroundService (ABS) performs several operations that can be
+    // long running and may not respond to the exit flag being set immediately.
+    // So, skip joining ABS to avoid blocking on these potential operations. It
+    // is safe to let ABS continue in the background, and ABS will stop if/when
+    // it finally checks the exit flag.
+    //
+    // accounts_background_service.join().unwrap();
```

Comment:
Ahh ok, I think I misunderstood your comment here: https://github.com/anza-xyz/agave/issues/346#issuecomment-2147610699

Reply:
> I believe I just hadn't actually thought through all the implications in my comment. Sorry about that!

No worries! Glad we caught it in review!

I ended up doing a little more rework; instead of passing in a flag for whether to join, I return the thread handle and allow the caller to join if they so choose. Let me know what you think. I stuffed the result of `load_and_process_ledger()` into a struct as well, this is probably better incase we grow it more in the future